examples/http_auth.rs

```diff
@@ -0,0 +1,276 @@
+extern crate byteorder;
+extern crate env_logger;
+#[macro_use]
+extern crate error_chain;
+extern crate extprim;
+extern crate futures;
+extern crate hyper;
+extern crate log;
+extern crate mtproto;
+extern crate rand;
+extern crate select;
+extern crate serde;
+extern crate serde_mtproto;
+extern crate tokio_core;
+
+
+use std::str;
+
+use byteorder::{ByteOrder, BigEndian};
+use futures::{Future, Stream};
+use mtproto::tl::dynamic::TLObject;
+use mtproto::rpc::{AppInfo, Session};
+use mtproto::rpc::encryption::asymm;
+use mtproto::rpc::message::{Message, MessageType};
+use mtproto::schema;
+use rand::Rng;
+use select::document::Document;
+use select::predicate::Name;
+use serde::Serialize;
+use serde::de::DeserializeOwned;
+use tokio_core::reactor::{Core, Handle};
+
+
+mod error {
+    error_chain! {
+        links {
+            MtProto(::mtproto::Error, ::mtproto::ErrorKind);
+            SerdeMtProto(::serde_mtproto::Error, ::serde_mtproto::ErrorKind);
+        }
+
+        foreign_links {
+            Hyper(::hyper::Error);
+            Io(::std::io::Error);
+            SetLogger(::log::SetLoggerError);
+            Utf8(::std::str::Utf8Error);
+        }
+
+        errors {
+            NonceMismatch(expected: ::extprim::i128::i128, found: ::extprim::i128::i128) {
+                description("nonce mismatch")
+                display("nonce mismatch (expected {}, found {})", expected, found)
+            }
+
+            HtmlErrorText(error_text: String) {
+                description("RPC returned an HTML error")
+                display("RPC returned an HTML error with text: {}", error_text)
+            }
+
+            BadMessage(found_len: usize) {
+                description("Message is not HTML error and is < 24 bytes long")
+                display("Message is not HTML error and is {} < 24 bytes long", found_len)
+            }
+
+            UnknownHtmlErrorStructure(html: String) {
+                description("Unknown HTML error structure")
+                display("Unknown HTML error structure:\n{}", html)
+            }
+        }
+    }
+}
+
+use error::{ErrorKind, ResultExt};
+
+macro_rules! bailf {
+    ($e:expr) => {
+        return Box::new(futures::future::err($e.into()))
+    }
+}
+
+macro_rules! tryf {
+    ($e:expr) => {
+        match { $e } {
+            Ok(v) => v,
+            Err(e) => bailf!(e),
+        }
+    }
+}
+
+
+fn auth(handle: Handle) -> Box<Future<Item = (), Error = error::Error>> {
+    let app_info = tryf!(AppInfo::read_from_toml_file("AppInfo.toml")
+        .chain_err(|| "this example needs a AppInfo.toml file with `api_id` and `api_hash` fields in it"));
```

Comment:
hmm, any reason why not `dotenv`? that's what I'd been using.

Reply:
@habnabit I think `dotenv` is an application-level utility since it only changes environment but doesn't directly help with initializing `AppInfo`.

For library usage I prefer `envy` for its type-safe nature, though I can still leverage `dotenv` for examples nevertheless, thanks for the idea!